cpp/order/SolidLiquid.h

```diff
@@ -96,12 +97,11 @@ class SolidLiquid
     //! Returns a vector containing the size of all clusters.
     std::vector<unsigned int> getClusterSizes() const
     {
-        std::vector<unsigned int> sizes;
         auto keys = m_cluster.getClusterKeys();
-        for (auto cluster = keys.begin(); cluster != keys.end(); cluster++)
-        {
-            sizes.push_back(cluster->size());
-        }
+        std::vector<unsigned int> sizes;
+        sizes.reserve(keys.size());
+        std::transform(keys.begin(), keys.end(), std::back_inserter(sizes),
+                       [](auto& key) { return key.size(); });
```

Comment:
Just initialize with the correct size (instead of reserving after construction), then use `.begin()` as the output iterator.
```suggestion
        std::vector<unsigned int> sizes(keys.size());
        std::transform(keys.cbegin(), keys.cend(), sizes.begin(),
                       [](const auto& key) { return key.size(); });
```

Reply:
I chose this method since it's marginally more efficient, but that difference is almost certainly negligible in this case. I'm going to leave it since it's already written this way, but signing off on this change for future uses.